src/OpenFeature.Contrib.Providers.Flagd/DependencyInjection/FeatureBuilderExtensions.cs

```diff
@@ -0,0 +1,85 @@
+        return builder.AddProvider(sp => CreateProvider(sp, null));
+    }
+
+    /// <summary>
+    /// Adds the <see cref="FlagdProvider"/> to the <see cref="OpenFeatureBuilder"/> with <see cref="FlagdProviderOptions"/> configuration.
+    /// </summary>
+    /// <param name="builder">The <see cref="OpenFeatureBuilder"/> instance to configure.</param>
+    /// <param name="options">Options to configure <see cref="FlagdProvider"/>.</param>
+    /// <returns>The <see cref="OpenFeatureBuilder"/> instance for chaining.</returns>
+    public static OpenFeatureBuilder AddFlagdProvider(this OpenFeatureBuilder builder, Action<FlagdProviderOptions> options)
+    {
+        builder.Services.Configure(FlagdProviderOptions.DefaultName, options);
+        return builder.AddProvider(sp => CreateProvider(sp, null));
+    }
+
+    /// <summary>
+    /// Adds the <see cref="FlagdProvider"/> to the <see cref="OpenFeatureBuilder"/> with a specific domain and default <see cref="FlagdProviderOptions"/> configuration.
+    /// </summary>
+    /// <param name="builder">The <see cref="OpenFeatureBuilder"/> instance to configure.</param>
+    /// <param name="domain">The unique domain of the provider.</param>
+    /// <returns>The <see cref="OpenFeatureBuilder"/> instance for chaining.</returns>
+    public static OpenFeatureBuilder AddFlagdProvider(this OpenFeatureBuilder builder, string domain)
+    {
+        builder.Services.AddOptions<FlagdProviderOptions>(domain);
+        return builder.AddProvider(domain, CreateProvider);
+    }
+
+    /// <summary>
+    /// Adds the <see cref="FlagdProvider"/> to the <see cref="OpenFeatureBuilder"/> with a specific domain and <see cref="FlagdProviderOptions"/> configuration.
+    /// </summary>
+    /// <param name="builder">The <see cref="OpenFeatureBuilder"/> instance to configure.</param>
+    /// <param name="domain">The unique domain of the provider.</param>
+    /// <param name="options">Options to configure <see cref="FlagdProvider"/>.</param>
+    /// <returns>The <see cref="OpenFeatureBuilder"/> instance for chaining.</returns>
+    public static OpenFeatureBuilder AddFlagdProvider(this OpenFeatureBuilder builder, string domain, Action<FlagdProviderOptions> options)
+    {
+        builder.Services.Configure(domain, options);
+        return builder.AddProvider(domain, CreateProvider);
+    }
+
+    private static FlagdProvider CreateProvider(IServiceProvider provider, string domain)
+    {
+        var optionsMonitor = provider.GetRequiredService<IOptionsMonitor<FlagdProviderOptions>>();
+        var logger = provider.GetService<ILogger<FlagdProvider>>();
+        logger ??= NullLogger<FlagdProvider>.Instance;
+
+        FlagdProviderOptions options;
+        if (string.IsNullOrEmpty(domain))
+        {
+            options = optionsMonitor.Get(FlagdProviderOptions.DefaultName);
+        }
+        else
+        {
+            options = optionsMonitor.Get(domain);
+        }
```

Comment:
One small suggestion: consider replacing `string.IsNullOrEmpty(domain)` with `string.IsNullOrWhiteSpace(domain)`. 
Also, you can simplify this code by replacing the if/else block with a single line using the ternary operator.

So it would become:
```csharp
// Select options based on domain
var options = string.IsNullOrWhiteSpace(domain)
    ? optionsMonitor.Get(FlagdProviderOptions.DefaultName)
    : optionsMonitor.Get(domain);
```